src/Compilers/Core/Portable/CodeGen/ILBuilderEmit.cs

```diff
@@ -66,6 +66,11 @@ internal void EmitModuleVersionIdStringToken()
             this.GetCurrentWriter().WriteUInt32(Cci.MetadataWriter.ModuleVersionIdStringToken);
         }
 
+        internal void EmitSourceDocumentIndexToken(Cci.DebugSourceDocument document)
+        {
+            this.GetCurrentWriter().WriteUInt32((module?.GetFakeSourceDocumentTokenForIL(document) ?? 0xFFFF) | Cci.MetadataWriter.SourceDocumentIndex);
```

Comment:
Perhaps add a named constant for `0xFFFF` since the value is repeated several times and since a named value might make it clear when the value is needed.